cfme/utils/appliance/__init__.py

```diff
@@ -1474,26 +1473,26 @@ def reboot(self, wait_for_web_ui=True, log_callback=None):
         wait_for(lambda: client.uptime() < old_uptime, handle_exception=True,
             num_sec=600, message='appliance to reboot', delay=10)
 
-        if wait_for_web_ui:
-            self.wait_for_web_ui()
+        if wait_for_miq_ready:
+            self.wait_for_miq_ready()
 
     @logger_wrap("Waiting for web_ui: {}")
-    def wait_for_web_ui(self, timeout=900, running=True, log_callback=None):
-        """Waits for the web UI to be running / to not be running
+    def wait_for_miq_ready(self, timeout=900, log_callback=None):
+        """Waits for the web UI and API server to be ready / to not ready
 
         Args:
             timeout: Number of seconds to wait until timeout (default ``600``)
-            running: Specifies if we wait for web UI to start or stop (default ``True``)
-                     ``True`` == start, ``False`` == stop
         """
-        prefix = "" if running else "dis"
-        (log_callback or self.log.info)('Waiting for web UI to ' + prefix + 'appear')
-        result, wait = wait_for(self._check_appliance_ui_wait_fn, num_sec=timeout,
-            fail_condition=not running, delay=10)
+        num_secs = _get_timeout_secs(timeout)
+
+        (log_callback or self.log.info)('Waiting for web UI to appear')
+        result, secs_taken = wait_for(self._check_appliance_ui_wait_fn,
+                                num_sec=num_secs, fail_condition=False, delay=10)
+        self.wait_for_api_available(num_secs - secs_taken)
```

Comment:
What is your reasoning for waiting for the reduced time `num_secs - secs_taken`? Thoughts on having two timeouts, one for the UI and one for the API?

Reply:
My reasoning is to keep the maximum total time the function is blocking (and what we allow CFME to take) the same while not introducing too much complications.

Introducing second timeout of course possible, but I don't see any value of it, nor I can imagine useful implementation without adding uncertainty on how much time is in the end each of the service (UI/API) given.

Note I was considering using https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.wait
to poll the services in parallel. I am not sure how thread-safe is the code that would have to be called in the `concurrent.future`. This would be fun to play with though.

Perhaps some kind of `wait_for(lambda: are_both_services_running(), ...)` may be the way to go.

Reply:
That makes sense @jaryn, your point about keeping the total time blocking is a good one and I agree there. I think we should leave this as is